src/main/java/org/isf/medicalstock/rest/StockMovementController.java

```diff
@@ -0,0 +1,247 @@
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.isf.ward.model.Ward;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/stockmovements", produces = MediaType.APPLICATION_JSON_VALUE)
+public class StockMovementController {
+	@Autowired
+	private MovementMapper movMapper;
+	
+	@Autowired
+	private LotMapper lotMapper;
+	
+	@Autowired
+	private MovBrowserManager movManager;
+	
+	@Autowired
+	private MovStockInsertingManager movInsertingManager;
+	
+	@Autowired
+	private MedicalBrowsingManager medicalManager;
+	
+	/**
+	 * Insert a list of charging {@link Movement}s and related {@link Lot}s
+	 * 
+	 * @param movementDTOs - the list of {@link Movement}s
+	 * @param referenceNumber - the reference number to be set for all movements
+	 * 		   if {@link null}, each movements must have a different referenceNumber 
+	 * @return 
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/stockmovements/charge", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> newMultipleChargingMovements(@RequestBody List<MovementDTO> movementDTOs, 
+			@RequestParam(name="ref", required=true) String referenceNumber) throws OHServiceException {
+		ArrayList<Movement> movements = new ArrayList<Movement>();
+		movMapper.map2ModelList(movementDTOs).forEach(mov -> {
+			movements.add(mov);
+		});
+		boolean done = movInsertingManager.newMultipleChargingMovements(movements, referenceNumber);
+		return ResponseEntity.status(HttpStatus.CREATED).body(done);
+	}
+	
+	/**
+	 * Insert a list of discharging {@link Movement}s
+	 * 
+	 * @param movementDTOs - the list of {@link Movement}s
+	 * @param referenceNumber - the reference number to be set for all movements
+	 * 		   if {@link null}, each movements must have a different referenceNumber 
+	 * @return 
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/stockmovements/discharge", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> newMultipleDischargingMovements(@RequestBody List<MovementDTO> movementDTOs, 
+			@RequestParam(name="ref", required=true) String referenceNumber) throws OHServiceException {
+		ArrayList<Movement> movements = new ArrayList<Movement>();
+		movMapper.map2ModelList(movementDTOs).forEach(mov -> {
+			movements.add(mov);
+		});
```

Comment:
We probably avoid foreach iteration assigning movMapper.map2ModelList(movementDTOs) to movements?